<launch>

  <node pkg="occgrid_planner" type="occgrid_planner_b" name="planner" output="screen">
    <param name="neighbourhood" value="8"/>
    <param name="base_frame" value="/body"/>
    <param name="debug" value="False"/>
    <remap from="/planner/occ_grid" to="/map"/>
    <remap from="/planner/goal" to="/move_base_simple/goal"/>
  </node>


  <node pkg="occgrid_planner" type="path_optimizer_b" name="path_optimizer" output="screen">
      <param name="max_acceleration" value="0.3"/>
      <param name="max_braking" value="0.1"/>
      <param name="velocity" value="0.1"/>
      <remap from="/path_optimizer/path" to="/planner/path"/>
  </node>

  <node pkg="occgrid_planner" type="path_follower_b" name="path_follower" output="screen">
      <param name="Kx" value="0.5"/>
      <param name="Ky" value="0.5"/>
      <param name="Ktheta" value="1.0"/>
      <param name="max_rot_speed" value="1.0"/>
      <param name="max_velocity" value="0.2"/>
      <param name="max_y_error" value="1.0"/>
      <param name="look_ahead" value="1.0"/>
      <param name="base_frame" value="/body"/>
      <remap from="/path_follower/traj" to="/path_optimizer/trajectory"/>
      <remap from="/path_follower/twistCommand" to="/mux/autoCommand"/>
  </node>

<node pkg="obstacle_avoidance" type="obstacle_avoidance" name="obstacle_avoidance" output="screen">
    launch-prefix="xterm -e gdb --args">
    <param name="base_frame" value="/body" />
    <param name="max_range" value="5.0" />
    <param name="max_linear_velocity" value="1.0" />
    <param name="max_angular_velocity" value="1.0" />
    <param name="max_linear_accel" value="5.0" />
    <param name="max_angular_accel" value="5.0" />
    <param name="map_resolution" value="0.05" />
    <param name="linear_velocity_resolution" value="0.05" />
    <param name="angular_velocity_resolution" value="0.05" />
    <param name="robot_radius" value="0.1" />
    <param name="time_horizon" value="2.0" />
    <param name="k_v" value="10.0" />
    <param name="k_w" value="1.0" />

    <remap from="/obstacle_avoidance/scans" to="/vrep/hokuyoSensor"/>
    <remap from="/obstacle_avoidance/current_velocity" to="/vrep/twistStatus"/>
    <!-- <remap from="/obstacle_avoidance/command_velocity" to="/mux/twistCommand"/> -->
    <remap from="/obstacle_avoidance/output_velocity" to="/vrep/twistCommand"/> -->
</node>

  <node pkg="topic_tools" type="mux" name="cmd_mux"
      args="/obstacle_avoidance/command_velocity /teleop/twistCommand /mux/autoCommand">
  </node>

  <node pkg="vrep_ros_teleop" type="teleop_mux.py" name="teleop_mux">
      <param name="joystick_button" value="0" type="int"/>
      <param name="joystick_topic" value="/teleop/twistCommand"/>
      <param name="auto_button" value="1" type="int"/>
      <param name="auto_topic" value="/mux/autoCommand"/>
      <remap from="/teleop_mux/joy" to="/joy"/>
  </node>



</launch>
